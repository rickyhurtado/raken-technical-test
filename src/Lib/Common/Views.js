import React from 'react';
import { Link } from 'react-router-dom';
import { Helmet } from 'react-helmet';

export const PageTitle = ({ ...props }) => {
  let siteName = process.env.REACT_APP_SITE_NAME;
  let { title, customTitle, appName } = props;

  if (appName) {
    siteName = `${appName} | ${siteName}`;
  }

  let headTitle = customTitle ? title : `${title} | ${siteName}`;

  if (!title) {
    title = siteName;
    headTitle = siteName;
  }

  return (
    <h1 className="page-title">
      <Helmet title={headTitle} />
      {title}
    </h1>
  );
};

export const NavLink = ({ ...props }) => {
  const { count } = { ...props};
  let badge;

  if (count) {
    badge = (
      <span className="notification-badge">{count > 99 ? '99+' : count}</span>
    );
  }

  return (
    <li className={navLinkIsActive({ ...props })}>
      <Link to={props.to} onClick={closeNavbar}>{props.title} {badge}</Link>
    </li>
  );
};

export function viewportDimension() {
  const w = window,
        d = document,
        e = d.documentElement,
        g = d.getElementsByTagName('body')[0],
        width = w.innerWidth || e.clientWidth || g.clientWidth,
        height = w.innerHeight||e.clientHeight||g.clientHeight;

  return { width, height };
}

function navLinkIsActive({ ...props }) {
  return props.path === props.to ? 'active' : '';
}

function closeNavbar() {
  const { width } = viewportDimension();

  if (width < 768) {
    document.getElementById('js-navbar-toggle-btn').click();
  }
}
