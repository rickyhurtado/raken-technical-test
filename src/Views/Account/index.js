import MyProfile from './MyProfile';
import MyProjects from './MyProjects';
import Notifications from './Notifications';
import Tasks from './Tasks';

export default {
  MyProfile,
  MyProjects,
  Notifications,
  Tasks
};
