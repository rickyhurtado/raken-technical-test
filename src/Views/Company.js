import React, { Component } from 'react';
import { PageTitle } from '../Lib/Common/Views';

export default class Company extends Component {
  render() {
    return (
      <div className="page">
        <PageTitle title="Company" />
      </div>
    );
  }
};
