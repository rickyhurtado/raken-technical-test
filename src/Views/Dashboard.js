import React, { Component } from 'react';
import { PageTitle } from '../Lib/Common/Views';

export default class Dashboard extends Component {
  render() {
    return (
      <div className="page">
        <PageTitle title="Dashboard" />
      </div>
    );
  }
};
