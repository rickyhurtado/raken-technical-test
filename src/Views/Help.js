import React, { Component } from 'react';
import { PageTitle } from '../Lib/Common/Views';

export default class Help extends Component {
  render() {
    return (
      <div className="page">
        <PageTitle title="Help" />
      </div>
    );
  }
};
