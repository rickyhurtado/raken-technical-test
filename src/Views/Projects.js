import React, { Component } from 'react';
import { PageTitle } from '../Lib/Common/Views';

export default class Projects extends Component {
  render() {
    return (
      <div className="page">
        <PageTitle title="Projects" />
      </div>
    );
  }
};
