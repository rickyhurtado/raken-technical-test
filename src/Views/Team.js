import React, { Component } from 'react';
import { PageTitle } from '../Lib/Common/Views';

export default class Team extends Component {
  render() {
    return (
      <div className="page">
        <PageTitle title="Team" />
      </div>
    );
  }
};
