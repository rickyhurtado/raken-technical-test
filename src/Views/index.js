import Account from './Account';
import Company from './Company';
import Dashboard from './Dashboard';
import Help from './Help';
import PageNotFound from './PageNotFound';
import Projects from './Projects';
import Team from './Team';

export default {
  Account,
  Company,
  Dashboard,
  Help,
  PageNotFound,
  Projects,
  Team
};
