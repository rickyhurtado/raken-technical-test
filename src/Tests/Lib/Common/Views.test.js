import React from 'react';
import { BrowserRouter, Route } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { PageTitle, NavLink } from '../../../Lib/Common/Views';

it('PageTitle component should render', () => {
  const component = renderer.create(
    <PageTitle title="Page Title" />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

it('PageTitle component should render with parent page title', () => {
  const component = renderer.create(
    <PageTitle title="Page Title" appName="Parent Page Title" />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

it('PageTitle component should render with customTitle prop', () => {
  const component = renderer.create(
    <PageTitle title="Page Title" customTitle />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

it('NavLink component should render without "active" class and notification badge', () => {
  const component = renderer.create(
    <BrowserRouter>
      <Route>
        <NavLink title="Test Page" to="/test-page" path="/current-page" />
      </Route>
    </BrowserRouter>
  );

  expect(component.toJSON()).toMatchSnapshot();
});

it('NavLink component should render with "active" class', () => {
  const component = renderer.create(
    <BrowserRouter>
      <Route>
        <NavLink title="Current Page" to="/current-page" path="/current-page" />
      </Route>
    </BrowserRouter>
  );

  expect(component.toJSON()).toMatchSnapshot();
});

it('NavLink component should render with notification badge', () => {
  const component = renderer.create(
    <BrowserRouter>
      <Route>
        <NavLink title="Current Page" to="/current-page" path="/current-page" count="7" />
      </Route>
    </BrowserRouter>
  );

  expect(component.toJSON()).toMatchSnapshot();
});
