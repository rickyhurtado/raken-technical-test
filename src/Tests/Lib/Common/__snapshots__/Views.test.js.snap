// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink component should render with "active" class 1`] = `
<li
  className="active"
>
  <a
    href="/current-page"
    onClick={[Function]}
  >
    Current Page
     
  </a>
</li>
`;

exports[`NavLink component should render with notification badge 1`] = `
<li
  className="active"
>
  <a
    href="/current-page"
    onClick={[Function]}
  >
    Current Page
     
    <span
      className="notification-badge"
    >
      7
    </span>
  </a>
</li>
`;

exports[`NavLink component should render without "active" class and notification badge 1`] = `
<li
  className=""
>
  <a
    href="/test-page"
    onClick={[Function]}
  >
    Test Page
     
  </a>
</li>
`;

exports[`PageTitle component should render 1`] = `
<h1
  className="page-title"
>
  Page Title
</h1>
`;

exports[`PageTitle component should render with customTitle prop 1`] = `
<h1
  className="page-title"
>
  Page Title
</h1>
`;

exports[`PageTitle component should render with parent page title 1`] = `
<h1
  className="page-title"
>
  Page Title
</h1>
`;
